src/git-set-file-times.js

```diff
@@ -0,0 +1,63 @@
+import { getExecOutput } from '@actions/exec';
+import { existsSync, utimesSync } from 'fs';
+
+/**
+ * @returns {Promise<void>}
+ */
+async function gitSetFileTimes() {
+    return await ls().then(list => changeTime(list));
+}
+
+/**
+ * @returns {Promise<string[]>}
+ */
+async function ls() {
+    return new Promise(async resolve => {
+        const ls = [];
+        const output = await getExecOutput('git', ['ls-files', '-z'], { silent: true });
+        output.stdout.split('\n').forEach(line => {
+            ls.push(...line.split('\0').filter(f => !!f));
+        });
+        resolve(ls);
+    });
+}
+
+/**
+ * @param {string[]} files
+ * @returns {Promise<void>}
+ */
+async function changeTime(files) {
+    return new Promise(async resolve => {
+        const output = await getExecOutput('git', ['log', '-m', '-r', '--name-only', '--no-color', '--pretty=raw', '-z'], { silent: true });
+        let time = new Date();
+        output.stdout.split('\n').forEach(line => {
+            const m = line.match(/^committer .*? (\d+) (?:[\-\+]\d+)$/);
+            if (m) {
+                time = new Date(parseInt(m[1]) * 1000);
+                return;
+            }
+            const m1 = line.match(/(.+)\0\0commit [a-f0-9]{40}( \(from [a-f0-9]{40}\))?$/);
+            const m2 = line.match(/(.+)\0$/);
+            const list = (m1 ? m1[1] : m2 ? m2[1] : '').split(/\0/);
+            if (list.length <= 0) {
+                return;
+            }
+            list.forEach(file => {
+                if (!file) {
+                    return;
+                }
+                const index = files.indexOf(file);
+                if (index < 0) {
+                    return;
+                }
+                if (existsSync(file)) {
+                    utimesSync(file, time, time);
+                    files.splice(index, 1);
+                }
+            });
+        });
+        resolve();
+    });
+}
+
+export default gitSetFileTimes;
```

Comment:
This file does a lot of dark magic, but I trust that you've translated it carefully from Perl to have the same behavior :heart:. Following the trend of other review comments, I'd just `async`ify `existsSync` and `utimesSync` to the `access` and `utimes` functions you can find on `fs/promises`.

I'd also rename this file to `git_set_file_times.js`, to be more consistent with the naming of other JavaScript files in the repository.

Reply:
Fix by c7f34d2e8097b62ba4dc97c22d86fb667c5f3cbf